src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,211 @@
+            statement.setLong(1, userId);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            logger.error("Can't delete role with id" + userId, e);
+        }
+    }
+
+    public Set<Role> getRolesDb(User user) {
+        Set<Role> roles = new HashSet<>();
+        String query = "SELECT * FROM roles INNER JOIN users_roles"
+                + " USING (role_id) WHERE user_id = ?;";
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, user.getId());
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Long roleId = resultSet.getLong("role_id");
+                String roleDb = resultSet.getString("role_name");
+                Role role = Role.of(roleDb);
+                roles.add(role);
+                return roles;
+            }
+        } catch (SQLException e) {
+            logger.error("Can’t get role for user with login = " + user.getLogin(), e);
+        }
+        return roles;
+    }
+
+    @Override
+    public User add(User user) {
+        String query = "INSERT INTO users (name, surname, password, login, token)"
+                + " VALUES (?, ?, ?, ?, ?);";
+        try (PreparedStatement statement
+                     = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getSurname());
+            statement.setString(3, user.getPassword());
+            statement.setString(4, user.getLogin());
+            statement.setString(5, user.getToken());
+            statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            generatedKeys.next();
+            Long userId = generatedKeys.getLong(1);
+            user.setId(userId);
+            user.addRole(Role.of("USER"));
+            addRoleToDB(user);
+            return user;
+        } catch (SQLException e) {
+            logger.error("Can't add user with login = " + user.getLogin(), e);
+        }
+        return user;
+    }
+
+    @Override
+    public User get(Long id) {
+        String query = "SELECT * FROM users WHERE user_id = ?;";
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                String name = resultSet.getString("name");
+                String surname = resultSet.getString("surname");
+                String token = resultSet.getString("token");
+                String login = resultSet.getString("login");
+                String password = resultSet.getString("password");
+                User user = new User(name, surname, login, password);
+                user.setId(id);
+                user.setToken(token);
```

Comment:
@TatianaPirch is this code repeatable in several places in this class? if yes - is it possible move it to separate method?